### PR TITLE
[aide] Add Aide support

### PR DESCRIPTION
**Pull request by adam-vest**
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/aide.py
Comment (jcastill) [inline sos/report/plugins/aide.py:39]: This will generate an empty output and empty file if there are no inconsistencies in the database, right?

**Review by jcastill — APPROVED** [review]
Looks good in RHEL

File: sos/report/plugins/aide.py
Comment (adam-vest) [reply sos/report/plugins/aide.py:39]: Hey! It's only doing a linting of the config file, not anything with the database (that's `--check, -C`), but you are right that if the config file has no syntax errors then the output and resultant file will be empty.

**Comment by arif-ali** [comment]
It would be neater for the `conf_file` to be defined in the respective classes, and then call the command to collect in the main cleass? wdyt?

```python
self.add_cmd_output(f"aide -c {self.conf_file} --config-check")
```

**Comment by adam-vest** [comment]
Hey Arif! Do you mean like:
```
class Aide(Plugin):
...
    def setup(self):
        self.add_cmd_output(f"aide -c {self.conf_file} --config-check")
...
class RedHatAide(Aide, RedHatPlugin):
    conf_file = "/etc/aide.conf"
...
class DebianAide(Aide, DebianPlugin, UbuntuPlugin):
    conf_file = "/etc/aide/aide.conf"
```
I can see a certain logic behind going that route. It's one more line of code, but I think I do agree that it can be neater split up that way.

**Comment by arif-ali** [comment]
@adam-vest yeah, something like that.

This also means that if we add more command collections in the future, we just add/update it to the common class, and it can then manage both scenarios

File: sos/report/plugins/aide.py
Comment (TurboTurtle) [inline sos/report/plugins/aide.py:46]: flake8 linting check is failing here with a whitespace.

**Review by TurboTurtle — COMMENTED** [review]
LGTM after the flake8 alert is resolved. My only other comment would have also been to set `conf_file` as a class attr. Glad that's already been done.

File: sos/report/plugins/aide.py
Comment (adam-vest) [reply sos/report/plugins/aide.py]: Gah :man_facepalming: missed that tiny little squiggly in vscode. Fixed!

**Comment by adam-vest** [comment]
Not sure why the tests failed, but I'm guessing it has something to do with this line:
```
self.add_cmd_output(f"aide -c {self.conf_file} --config-check")
```

I ran that on a Bionic machine and it seemed to work fine? It looks like the standard way that other plugins might do this would be like:
```
self.add_cmd_output("aide -c %s --config-check" % self.conf_file)
```

...which I've confirmed also works fine. Would that satisfy whatever check failed?

**Comment by arif-ali** [comment]
@adam-vest I re-ran the test, and it went through without any problems, I think it was a one-off. I checked the previous runs too, and it had gone through, and was unrelated to your change.

Final thing is, if you got squash your commits then that would be awesome

**Comment by adam-vest** [comment]
Ugh, I'm making it worse. idk how to "squash" these commits properly. I'll try to fix whatever I just did.

**Comment by adam-vest** [comment]
OK...I think/hope that's all cleaned up now...sorry for all the trouble!

**Comment by TurboTurtle** [comment]
I'm not entirely sure what, but something seems to have gone sideways as the second commit pulls in changes that are already in `main`.

Luckily this PR allows maintainer edits, so I should be able to fix this up on my end and force-push over your branch, and then merge this. I've not tried a force-push on a fork branch before though, so hopefully that works. If not I'll write out the steps to fix this so you can do them locally. Give me a little bit to finish up some work stuff and I'll get to straightening this out later today.

**Comment by TurboTurtle** [comment]
Ok, looks like pushing to your fork worked. :tada: 

I think everything looks fine now, we'll let the CI run and then just need an ack from @arif-ali or @pmoravec.

**Comment by adam-vest** [comment]
Good gravy...thanks for taking care of that @TurboTurtle , and sorry again heh

**Comment by TurboTurtle** [comment]
No worries, at the end of the day sometimes it boils down to Git Is Weird(tm).

File: sos/report/plugins/aide.py
Comment (pmoravec) [inline sos/report/plugins/aide.py:42]: We might add `add_forbidden_path` for `logrotate` plugin for this file to prevent both plugins to collect this file, but.. assuming this file won't ever rise to MBs in size, there is no technical need.

**Review by arif-ali — APPROVED** [review]

**Pull request merged**
